src/Modules/UngradedModules/SubquoModule.jl

```diff
@@ -1619,6 +1619,130 @@ function intersect(M::SubquoModule{T}, N::SubquoModule{T}) where T
   throw(ArgumentError("Relations of M and N are not equal."))
 end
 
+########################################
+# module quotients
+########################################
+
+@doc raw"""
+     annihilator(N::SubquoModule{T}) where T
+
+Return the annihilator of `N`.
+
+!!! note
+    By definition, the annihilator of $N$ is the ideal $0:N = \{a \in R \mid aN = 0\}$. 
+    Here, `R = base_ring(N)`.
+
+# Examples
+
+```jldoctest
+julia> R, (x, y, z) = polynomial_ring(QQ, [:x, :y, :z]);
+
+julia> F = free_module(R, 1);
+
+julia> AN = R[y;];
+
+julia> BN = R[x^2; y^3; z^4];
+
+julia> N = SubquoModule(F, AN, BN)
+Subquotient of Submodule with 1 generator
+1 -> y*e[1]
+by Submodule with 3 generators
+1 -> x^2*e[1]
+2 -> y^3*e[1]
+3 -> z^4*e[1]
+
+julia> J = annihilator(N)
+Ideal generated by
+  y^2
+  x^2
+  z^4
+
+```
+"""
+function annihilator(M::SubquoModule{T}) where T
+  R = base_ring(M)
+  F = FreeMod(R, 1)
+  I = ideal(R, [one(R)])
+  for v in gens(M)
+    h = hom(F, M, [v])
+    K, _ = kernel(h)
+    # this is a hack, because getindex is broken for K
+    g = Oscar.as_matrix(F, ambient_representatives_generators(K))
+    I = intersect(I, ideal(R, minors(g, 1)))
+  end
+  return I
+end
+
+function annihilator(N::SubquoModule{T}) where T <: MPolyRingElem
+  R = base_ring(N)
+  N_quo = isdefined(N, :quo) ? N.quo : SubModuleOfFreeModule(ambient_free_module(N), Vector{elem_type(ambient_free_module(N))}())
+  A = N.sub
+  SA = singular_generators(A.gens) 
+  B = N_quo
+  SB = singular_generators(B.gens)
+  return MPolyIdeal(R, Singular.quotient(SB, SA))
+end
+
+@doc raw"""
+    quotient(M::SubquoModule{T}, N::SubquoModule{T}) where T
+
+Given subquotients `M` and `N` such that `ambient_module(M) == ambient_module(N)`,
+return the module quotient of `M` by `N` regarded as submodules of the common ambient module.
+
+Alternatively, use `M:N`.
+
+!!! note
+    By definition, $M:N = \{a \in R \mid aN \subset M\}$. Here, `R = base_ring(M) = base_ring(N)`.
+
+# Examples
+
+```jldoctest
+julia> R, (x, y, z) = graded_polynomial_ring(QQ, [:x, :y, :z]);
+
+julia> F = graded_free_module(R, 1);
+
+julia> B = R[x^2; y^3; z^4];
+
+julia> AM = R[x;];
+
+julia> M = SubquoModule(F, AM, B)
+Graded subquotient of submodule of F generated by
+1 -> x*e[1]
+by submodule of F generated by
+1 -> x^2*e[1]
+2 -> y^3*e[1]
+3 -> z^4*e[1]
+
+julia> AN = R[y;];
+
+julia> N = SubquoModule(F, AN, B)
+Graded subquotient of submodule of F generated by
+1 -> y*e[1]
+by submodule of F generated by
+1 -> x^2*e[1]
+2 -> y^3*e[1]
+3 -> z^4*e[1]
+
+julia> L = quotient(M, N)
+Ideal generated by
+  x
+  y^2
+  z^4
+
+```
+"""
+function quotient(M::SubquoModule{T}, N::SubquoModule{T}) where T
+  @assert base_ring(M) == base_ring(N)
+  @assert ambient_module(M) == ambient_module(N)
+  MplusN, iM, _ = sum(M, N)
+  Q, _ = quo(MplusN, [iM(x) for x in gens(M)])
+  return annihilator(Q)
+end
+
+(::Colon)(M::SubquoModule{T}, N::SubquoModule{T}) where T = quotient(M, N)
```

Comment:
I personally don't like overloading basic operators (like `:` to create `UnitRange`s) with more complex functions as this can make it really hard to understand code that uses these.
What do other people think of this?

Reply:
This is already in use for ideal quotients. It is the standard mathematical notation and, in interactve sessions, everybody would love to use it that way. Of course, when writing code, you can still use quotient.

Reply:
While I am also in general wary of this, I believe it is fine here, and extremely convenient if one does a certain kind of math.

And as @wdecker points out, we already do it, this just adds another case (which I think is good, it is more confusing if we support this *sometimes*  but not in all cases in which it makes sense)